src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2398,21 +2404,21 @@ internal static EventOpcode GetOpcodeWithDefault(EventOpcode opcode, string? eve
         /// </summary>
         private sealed class OverrideEventProvider : EventProvider
         {
-            public OverrideEventProvider(EventSource eventSource, EventProviderType providerType)
+            public OverrideEventProvider(Func<EventSource?> eventSourceFactory, EventProviderType providerType)
                 : base(providerType)
             {
-                this.m_eventSource = eventSource;
+                this.m_eventSourceFactory = eventSourceFactory;
```

Comment:
>this.m_eventSourceFactory [](http://example.com/codeflow?start=16&length=25)

nit: can we use the convention `_eventSourceFactory` instead of `m_eventSourceFactory`? and no need using `this.` too.